tq42/utils/utils.py

```diff
@@ -103,3 +106,30 @@ def dynamic_create_exp_run_request(
         **keyword_args,
     )
     return experiment_request
+
+
+def save_token(service_name: str, backup_save_path: str, token: str) -> str:
+    try:
+        save_location = "keyring"
+        keyring.set_password(
+            service_name=service_name,
+            username="username",
+            password=token,
+        )
+        return save_location
+
+    except (NoKeyringError, InitError):
+        print(f"Access token saving in keyring error. Saved to {backup_save_path}.")
```

Comment:
you are having this twice now, telling the user access token was saved. Especially since here you do not provide further information. You can remove this line 👍

Reply:
Also you do not know if it is an access token or any other token.

Reply:
you are right. I have removed it now.